domain/src/Services/TransactionDispatcher/Handlers/NftHandler.php

```diff
@@ -65,9 +68,18 @@ private function handleAcquisition(Transaction $transaction): void
         $nftId = NftId::fromNftId($transaction->receivedAsset);
         $nft = $this->nftRepository->get($nftId);
 
-        $nft->acquire(new AcquireNft(
-            date: $transaction->date,
-            costBasis: $transaction->marketValue->plus($this->splitFees($transaction)), // @phpstan-ignore-line
-        ));
+        if ($nft->isAlreadyAcquired()) {
+            $nft->increaseCostBasis(new IncreaseNftCostBasis(
+                date: $transaction->date,
+                costBasisIncrease: $transaction->marketValue->plus($this->splitFees($transaction)), // @phpstan-ignore-line
+            ));
+        } else {
+            $nft->acquire(new AcquireNft(
+                date: $transaction->date,
+                costBasis: $transaction->marketValue->plus($this->splitFees($transaction)), // @phpstan-ignore-line
+            ));
+        }
```

Comment:
ℹ️ Had completely missed the already-acquired NFT use case (when a NFT is minted from several other NFTs, for instance). This came up while testing the command using the sample transactions (also added as part of this PR)